gcloud/bigquery/dataset.py

```diff
@@ -0,0 +1,350 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Datasets."""
+
+import datetime
+
+import pytz
+import six
+
+from gcloud.exceptions import NotFound
+
+
+class Dataset(object):
+    """Datasets are containers for tables.
+
+    See:
+    https://cloud.google.com/bigquery/docs/reference/v2/datasets
+
+    :type name: string
+    :param name: the name of the dataset
+
+    :type client: :class:`gcloud.bigquery.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the dataset (which requires a project).
+    """
+
+    def __init__(self, name, client):
+        self.name = name
+        self._client = client
+        self._properties = {}
+
+    @property
+    def project(self):
+        """Project bound to the dataset."""
+        return self._client.project
+
+    @property
+    def path(self):
+        """URL path for the dataset's APIs"""
+        return '/projects/%s/datasets/%s' % (self.project, self.name)
+
+    @property
+    def created(self):
+        """Datetime at which the dataset was created.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        """
+        return _datetime_from_prop(self._properties.get('creationTime'))
+
+    @property
+    def dataset_id(self):
+        """ID for the dataset resource.
+
+        :rtype: string, or ``NoneType``
+        """
+        return self._properties.get('id')
+
+    @property
+    def etag(self):
+        """ETag for the dataset resource.
+
+        :rtype: string, or ``NoneType``
+        """
+        return self._properties.get('etag')
+
+    @property
+    def modified(self):
+        """Datetime at which the dataset was last modified.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        """
+        return _datetime_from_prop(self._properties.get('lastModifiedTime'))
+
+    @property
+    def self_link(self):
+        """URL for the dataset resource.
+
+        :rtype: string, or ``NoneType``
+        """
+        return self._properties.get('selfLink')
+
+    @property
+    def default_table_expiration_ms(self):
+        """Default expiration time for tables in the dataset.
+
+        :rtype: integer, or ``NoneType``
```

Comment:
Regarding: "It would be redundant with the summary line."

I agree, but it's not uncommon for parts of a Sphinx docstring to be redundant.